core/lib/dal/src/via_votes_dal.rs

```diff
@@ -165,4 +165,82 @@ impl ViaVotesDal<'_, '_> {
 
         Ok(row.max_batch_number.map(|n| n as u32))
     }
+
+    pub async fn verify_votable_transaction(
+        &mut self,
+        l1_batch_number: u32,
+        tx_id: H256,
+        l1_batch_status: bool,
+    ) -> DalResult<()> {
+        sqlx::query!(
+            r#"
+            UPDATE via_votable_transactions
+            SET
+                is_verified = TRUE,
+                l1_batch_status = $3,
+                updated_at = NOW()
+            WHERE
+                l1_batch_number = $1
+                AND tx_id = $2
+            "#,
+            l1_batch_number as i64,
+            tx_id.as_bytes(),
+            l1_batch_status
+        )
+        .instrument("verify_transaction")
+        .execute(self.storage)
+        .await?;
+        Ok(())
+    }
+
+    pub async fn get_first_not_finilized_block(&mut self) -> DalResult<Option<i64>> {
+        let l1_block_number = sqlx::query_scalar!(
+            r#"
+            SELECT
```

Comment:
Maybe a more optimised and intuitive query would be:

`SELECT
    MIN(l1_batch_number) as "l1_batch_number"
  FROM
    via_votable_transactions
  WHERE
    is_finalized = FALSE
`

Reply:
fixed